code/controllers/subsystem/atoms.dm

```diff
@@ -108,22 +108,25 @@ SUBSYSTEM_DEF(atoms)
 	BadInitializeCalls = SSatoms.BadInitializeCalls
 
 /datum/controller/subsystem/atoms/proc/setupGenetics()
-	var/list/avnums = new /list(DNA_STRUC_ENZYMES_BLOCKS)
-	for(var/i=1, i<=DNA_STRUC_ENZYMES_BLOCKS, i++)
-		avnums[i] = i
-		CHECK_TICK
-
-	for(var/A in subtypesof(/datum/mutation/human))
-		var/datum/mutation/human/B = new A()
-		if(B.dna_block == NON_SCANNABLE)
+	var/list/mutations = subtypesof(/datum/mutation/human)
+	shuffle_inplace(mutations)
+	for(var/A in subtypesof(/datum/generecipe))
+		new A ()

```

Comment:
There's absolutely no need to create and delete these with current code, use initial. (If you want more than two mutations in these you need to change this a bit)

Reply:
No ?

Reply:
wtf are you on about time

Reply:
wait no fuck that was just my experience because the loop needed to be typeless and then cast as the /datum/generecipe